remote_test.go

```diff
@@ -334,27 +401,28 @@ func (s *RemoteSuite) TestFetchWithHashesButMissing(c *C) {
 	})
 
 	c.Assert(err.Error(), Equals, "empty packfile")
+	s.assertShallows(c, r, 0)
+}
+
+func (s *RemoteSuite) assertShallows(c *C, r *Remote, expected int) {
+	shallowCommits, err := r.s.Shallow()
+	c.Assert(err, IsNil)
+
+	c.Assert(shallowCommits, HasLen, expected)
```

Comment:
I extracted this because managing all these concerns inside of `testFetch` was getting to be abrasive.